element_calcium_imaging/imaging_report.py

```diff
@@ -0,0 +1,418 @@
+import cv2
+import numpy as np
+import datajoint as dj
+from functools import partial
+from plotly.io import from_json
+import plotly.graph_objects as go
+from ipywidgets import widgets as wg
+from . import imaging, scan
+
+schema = imaging.schema
+
+
+def single_to_3channel_image(image, low_q, high_q):
+    low_p, high_p = np.percentile(image, [low_q, high_q])
+    image = np.clip(image, low_p, high_p)
+
+    image -= image.min()
+    image *= 255 / image.max()
+    image = np.uint8(image)
+    image = np.dstack([image] * 3)
+    return image
+
+
+def paint_rois(image, mask_xpix, mask_ypix):
+    # Generate random hsv colors
+    SATURATION = 40
+    VALUE = 255
+    hues = np.random.sample(size=len(mask_xpix)) * 10000
+    hsv_colors = np.stack(
+        [hues, np.full(len(mask_xpix), SATURATION), np.full(len(mask_xpix), VALUE)]
+    ).T
+
+    # Assign colors to each region
+    masks = cv2.cvtColor(image, cv2.COLOR_RGB2HSV)
+    for xpix, ypix, hsv_color in zip(mask_xpix, mask_ypix, hsv_colors):
+        masks[ypix, xpix] = hsv_color
+
+    masks = np.uint8(cv2.cvtColor(masks.astype(np.float32), cv2.COLOR_HSV2RGB))
+    return masks
```

Comment:
Why construct the color array?

```suggestion
    # Assign colors to each region
    masks = cv2.cvtColor(image, cv2.COLOR_RGB2HSV)
    for xpix, ypix, hsv_color in zip(mask_xpix, mask_ypix, hsv_colors):
        masks[ypix, xpix] = [VALUE, SATURATION, np.random.random()]
    return np.uint8(cv2.cvtColor(masks.astype(np.float32), cv2.COLOR_HSV2RGB))
```

Reply:
But you may also consider preserving the underlying image:

```python
    # Assign colors to each region
    masks = cv2.cvtColor(image, cv2.COLOR_RGB2HSV)
    for xpix, ypix in zip(mask_xpix, mask_ypix):
        masks[ypix, xpix, 0:2] = [np.random.random(), SATURATION]
    return np.uint8(cv2.cvtColor(masks.astype(np.float32), cv2.COLOR_HSV2RGB))
```

Reply:
I do weighted averaging to construct the transparently overlayed masks on the original image. I will verify your suggestion to see the final output.

Reply:
Your suggestion is generating color per pixel instead of per mask.

Reply:
no, it's one color for the whole mask.

Reply:
How about this? The code below generates the exact same result on the left panel of Figure above and generates the colors within the for loop
```
def paint_rois2(image, mask_xpix, mask_ypix):
    # Generate random hsv colors
    SATURATION = 40
    VALUE = 255

    # Assign colors to each region
    masks = cv2.cvtColor(image, cv2.COLOR_RGB2HSV)
    for xpix, ypix in zip(mask_xpix, mask_ypix):
        masks[ypix, xpix] = [np.random.random()*255, SATURATION, VALUE]
    return np.uint8(cv2.cvtColor(masks.astype(np.float32), cv2.COLOR_HSV2RGB))
```

Reply:
Good. But these masks cover up the image behind them.  It would be great to keep the VALUE from the original image, random HUE, and 100% SATURATION.  This will allow you to see the mask and the image behind it.

Reply:
The transparency effect is achieved with the alpha_combine_2images function.

Reply:
Also, it does not appear useful to stretch the contrast so much by subtracting the mean. It hides a lot of useful information in the darker regions.